### PR TITLE
User story 18/see past drawings

### LINKED ISSUE #18: As a player, I want to see my past drawings, so that I remember a fun drawing.

### PR DESCRIPTION
**Pull request by BrentTB**
Major updates to the database were made
a game session is saved every time a game starts
every drawing that is submitted is saved to the database
The history page now shows (unformatted) drawings from each game

closes #18

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -6,8 +6,8 @@ const Router = require('./routes/router')
 
 const app = express()
 
-app.use(bodyParser.json())
-app.use(bodyParser.urlencoded({ extended: true }))
+app.use(bodyParser.json({ limit: '10mb' }))
```
Comment (RudolphTorlage) [inline src/app.js:9]: I'm assuming this is because the image data is very large?

File: src/controller/dbController.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 const db = require('../db/database')
+const dbAccess = require('../db/dbAccess')
```
Comment (RudolphTorlage) [inline src/controller/dbController.js:2]: New database file? Why not add code in the dbController?

File: src/db/dbAccess.js
Comment (RudolphTorlage) [inline src/db/dbAccess.js:1]: This could all go in the dbController

File: src/db/dbAccess.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  db.query(query, values, (error) => {
+    if (error) throw error
+  })
+}
+
+exports.newGame = (names) => {
+  return new Promise((resolve, reject) => {
+    const game = {}
+    game['gameName'] = 'some_default_game_name'
```
Comment (RudolphTorlage) [inline src/db/dbAccess.js:16]: Should implement room name upon room creation

File: src/server.js [hunk 4]
```diff
@@ -6,13 +6,35 @@ dotenv.config({ path: './config.env' })
...
 const rooms = {}
 const publicRooms = {}
 const rounds = {}
 const drawingSubmissions = {}
+const users = new Map()
+
+const fs = require('fs')
+
+// example function showing how to save the data to an image as a file
+async function test() {
```
Comment (RudolphTorlage) [inline src/server.js:22]: Does this code do anything?

File: src/server.js [hunk 5]
```diff
@@ -100,6 +122,16 @@ io.on('connection', (socket) => {
 
     drawingSubmissions[roomId][socket.id] = image
     updateGridSubmission(roomId, socket.id, 'drawing', image)
+    console.log('here:')
```
Comment (RudolphTorlage) [inline src/server.js:125]: Please remove unnecessary console logs

**Review by RudolphTorlage — CHANGES_REQUESTED** [review]
Please remove code in server file that saves data to local computer.
Please remove dbAccess file and localise all code to dbController.

**Review by RudolphTorlage — APPROVED** [review]
Thank you for addressing the requested changes.

File: src/controller/dbController.js (hunk 2 above)
Comment (BrentTB) [reply src/controller/dbController.js]: This is a very good point, and must be fixed for consistency

File: src/server.js (hunk 4 above)
Comment (BrentTB) [reply src/server.js]: no, it was a sample for saving images sent over a socket, but it must be removed as it is not needed

File: src/controller/dbController.js (hunk 2 above)
Comment (RudolphTorlage) [reply src/controller/dbController.js]: thank you for fixing this.

File: src/db/dbAccess.js (hunk 3 above)
Comment (RudolphTorlage) [reply src/db/dbAccess.js]: This was implemented in a future update.

**Pull request merged**